src/python/pants/init/extension_loader.py

```diff
@@ -136,22 +136,38 @@ def load_backend(build_configuration: BuildConfiguration.Builder, backend_packag
     :raises: :class:``pants.base.exceptions.BuildConfigurationError`` if there is a problem loading
       the build configuration.
     """
-    backend_module = backend_package + ".register"
+
+    if templating_args:
+        kwargs = {'backend_package': backend_package}
```

Comment:
Calling this arg `backend_package` I find slightly confusing, as in the configuration it's not really, but rather the name/alias used to connect the template section with the entry in the backend_pacakges list. So I see the logic here, but how do you feel about something like (trying to hold on to both concepts):
```suggestion
        kwargs = {'backend_package_alias': backend_package}
```